client.go

```diff
@@ -69,12 +70,50 @@ func NewClient(apiKey string, options ...ClientOption) (*Client, error) {
 	return &Client{client}, nil
 }
 
-func (c Client) Templates() (Templates, error) {
+func (c Client) TemplateByID(templateID string) (Template, error) {
+	var template Template
+	err := c.c.Get("./v2/template/" + templateID).JSON(&template).Error
+	return template, err
+}
+
+func (c Client) TemplateVersion(templateID string, version int) (Template, error) {
+	url := "/v2/template/" + templateID + "/version/" + strconv.Itoa(version)
+	var template Template
+	err := c.c.Get(url).JSON(&template).Error
+	return template, err
+}
+
+func (c Client) Templates(templateType string) (Templates, error) {
```

Comment:
@andrewoh531 can just use `typ` as argument name (is quite idiomatic to name `type` as `typ`).